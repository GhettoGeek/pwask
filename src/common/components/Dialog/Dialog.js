import React from 'react'
import PropTypes from 'prop-types'
import {
  Dialog, DialogTitle, DialogContent, DialogActions,
} from '@material-ui/core'

function EnhancedDialog({
  open, title, onClose, children, actions,
}) {
  return (
    <Dialog
      open={open}
      onClose={onClose}
      aria-labelledby="dialog-title"
      maxWidth="lg"
    >
      {title && (
        <DialogTitle id="dialog-title">
          {title}
        </DialogTitle>
      )}
      <DialogContent>{children}</DialogContent>
      {actions && (
        <DialogActions>{actions}</DialogActions>
      )}
    </Dialog>
  )
}

EnhancedDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  title: PropTypes.string,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  actions: PropTypes.node,
}

EnhancedDialog.defaultProps = {
  title: null,
  actions: null,
}

export default EnhancedDialog
