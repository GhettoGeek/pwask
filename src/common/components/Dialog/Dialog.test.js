import React from 'react'
import { shallow } from 'enzyme'
import Dialog from './Dialog'

describe('Dialog', () => {
  let mockFunction

  beforeEach(() => {
    mockFunction = jest.fn()
    mockFunction.mockReturnValue('test')
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly the component', () => {
    const wrapper = shallow(
      <Dialog
        open
        title="Title"
        onClose={mockFunction}
      >
        Lorem ipsum dolor sit amet
      </Dialog>,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
