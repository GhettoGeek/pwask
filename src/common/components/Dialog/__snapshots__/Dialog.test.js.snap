// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders correctly the component 1`] = `""`;
